--- a/werkzeug/useragents.py	2010-09-07 02:53:16.951944000 +0300
+++ b/werkzeug/useragents.py	2010-09-15 19:05:01.234595223 +0300
@@ -22,15 +22,16 @@
         (r'darwin|mac|os\s*x', 'macos'),
         ('win', 'windows'),
         (r'android', 'android'),
-        (r'x11|lin(\b|ux)?', 'linux'),
         ('(sun|i86)os', 'solaris'),
         (r'nintendo\s+wii', 'wii'),
         ('irix', 'irix'),
         ('hp-?ux', 'hpux'),
         ('aix', 'aix'),
         ('sco|unix_sv', 'sco'),
+        ('dragonfly', 'dragonfly'),
         ('bsd', 'bsd'),
-        ('amiga', 'amiga')
+        ('amiga', 'amiga'),
+        (r'x11|lin(\b|ux)?', 'linux'),
     )
     browsers = (
         ('googlebot', 'google'),
@@ -40,7 +41,8 @@
         (r'aol|america\s+online\s+browser', 'aol'),
         ('opera', 'opera'),
         ('chrome', 'chrome'),
-        ('firefox|firebird|phoenix|iceweasel', 'firefox'),
+        ('firefox|firebird|phoenix|iceweasel|namoroka|minefield', 'firefox'),
+        ('midoru', 'midori'),
         ('galeon', 'galeon'),
         ('safari', 'safari'),
         ('webkit', 'webkit'),
@@ -105,6 +107,7 @@
        -   `amiga`
        -   `android`
        -   `bsd`
+       -   `dragonfly`
        -   `hpux`
        -   `iphone`
        -   `irix`
@@ -125,6 +128,7 @@
         -   `camino`
         -   `chrome`
         -   `firefox`
+        -   `midori`
         -   `galeon`
         -   `google` *
         -   `kmeleon`
